src/libs/ajax/data-table-providers/WdsDataTableProvider.test.ts

```diff
@@ -691,6 +707,26 @@ describe('WdsDataTableProvider', () => {
   });
 });
 
+describe('updateRecord', () => {
+  it('restructures a WDS response', () => {
+    // Arrange
+    const provider = new TestableWdsProvider();
+
+    const expected: RecordAttributes = {
+      something: 123,
+      testAttr: 'string_value',
+    };
+
+    // Act
+    return provider
+      .updateRecord({ instance: uuid, recordName: 'test', recordId: 'record1', record: expected })
+      .then((actual) => {
+        // Assert
+        expect(actual.attributes).toStrictEqual(expected);
+      });
```

Comment:
Terra UI mostly uses async/await over Promise callbacks by convention.

```suggestion
    const actual = await provider.updateRecord({ instance: uuid, recordName: 'test', recordId: 'record1', record: expected });

    // Assert
    expect(actual.attributes).toStrictEqual(expected);
```
This suggest also requires marking the test function `async`.